plugins/policy/cache/cache_impl.go

```diff
@@ -143,8 +143,27 @@ func (pc *PolicyCache) LookupPodsByNSLabelSelector(policyNamespace string,
 // LookupPodsByLabelSelector evaluates label selector (expression and/or match
 // labels) and returns IDs of matching pods.
 func (pc *PolicyCache) LookupPodsByLabelSelector(
-	podLabelSelector *policymodel.Policy_LabelSelector) (pods []podmodel.ID) {
-	return nil
+	namespaceLabelSelector *policymodel.Policy_LabelSelector) (pods []podmodel.ID) {
+	// An empty namespace selector matches all namespaces.
+	if len(namespaceLabelSelector.MatchExpression) == 0 && len(namespaceLabelSelector.MatchLabel) == 0 {
+		allPods := pc.configuredPods.ListAll()
```

Comment:
check that namespaceLabelSelector is not nil first; if it is (possible if no label selectors are present in the policy), the check for  MatchExpression will panic.